geodepy/transform.py

```diff
@@ -506,6 +506,60 @@ def conform14(x, y, z, to_epoch, trans):
     return xtrans, ytrans, ztrans
 
 
+def mga94to20(zone, east, north, ell_ht=False):
+    """
+    Performs conformal transformation of Map Grid of Australia 1994 to Map Grid of Australia 2020 Coordinates
+    using the GDA2020 Tech Manual v1.2 7 parameter similarity transformation parameters
+    :param zone: Zone Number - 1 to 60
+    :param east: Easting (m, within 3330km of Central Meridian)
+    :param north: Northing (m, 0 to 10,000,000m)
+    :param ell_ht: Ellipsoid Height (m) (optional)
+    :return: MGA2020 Zone, Easting, Northing and Ellipsoid Height (if none provided, returns 0)
+    """
+    lat, lon, psf, gridconv = grid2geo(zone, east, north)
+    if ell_ht is False:
```

Comment:
This can be tidied up by taking everything except the setting of ell_ht out of the if statement. If you use ell_ht_in and ell_ht_out you can them both to zero if ell_ht is False.

Then you only need to do the other calls once (using ell_ht_in for llh2xyz and ell_ht_out for xyz2llh)

Reply:
Cool, I see where you're coming from. The downside is that there's an extra if-check after the conversion if no ellipsoid height is supplied so that this can be reset to zero, but it does make the code a bit shorter.

Reply:
Yeah, I was turning this over in my head after commenting. You could try

if ell_ht is False:
        x94, y94, z94 = llh2xyz(lat, lon, 0)
else:
        x94, y94, z94 = llh2xyz(lat, lon, ell_ht)

but then you still have to deal with the output.

I think what you got now is best